### PR TITLE
Main application structure

### PR DESCRIPTION
**Pull request by BaBrixx**
Closes #{19}

### Changes/Additions:
- Created the skeleton for the application including sidebars, main canvas, and navigation bars.

Big thanks to @NybyDK for his contributions! This would not have been possible without his knowledge!

### CODE REVIEW

**Comment by NybyDK** [comment]
Hvem end jeres Scrum Master er, consider yourself requested

**Review by KamyaPA — APPROVED** [review]

**Review by atjn — DISMISSED** [review]
Could you use some more descriptive names for the functions and variables? Everybody needs to be able to read this code and understand it.

What is `currentSide` referring to? The current side of what? Could you add a TS type that makes it clear which string types could be used here?

When reading `leftLength`, can you intuitively figure out that this refers to the width of the left side bar? What about `leftSidebarWidth` instead?

There are two functions `startExpand` and `stopExpand`. How is it clear that these are related to the function `resize`? Also what are they resizing/expanding?

I would also rename `e` to `event` to make it easier to read what is going on.

**Comment by NybyDK** [comment]
Feel free to fix! xd

File: src/routes/+page.svelte [hunk 1]
```diff
@@ -1,11 +1,120 @@
 <script lang="ts">
-	import { Hiking } from 'svelte-google-materialdesign-icons';
+	let currentSide: string = '';
+	let leftLength: number = 300;
+	let rightLength: number = 300;
+	let mainContainer: HTMLElement;
```
Comment (atjn) [inline src/routes/+page.svelte:5]: You don't need a reference to `mainContainer`, it should be possible to register the handlers on `document`, `window` or the bar itself. This is not something you need to change, just a suggestion :)

Comment (NybyDK) [reply src/routes/+page.svelte:5]: We did this on purpose so that when clicking outside of the main container, it didn't do the stuffs

**Review by AlbertHald — APPROVED** [review]
LGTM

File: src/routes/+page.svelte (hunk 1 above)
Comment (NybyDK) [reply src/routes/+page.svelte:11]: Resolved as executive decision by Bastian and Anders

**Review by atjn — DISMISSED** [review]
I think this is pretty good, but I still think the naming could be improved.

Does `currentSide` refer to the the panel being dragged, or the panel that is currently in focus, or the panel that the cursor is hovering over?

You use `PanelSide`, `Side`, and `Sidebar` to refer to the same thing. 

Enum is very nice!

**Review by atjn — DISMISSED** [review]
LVGTMTVMFYHW

**Review by christian140398 — COMMENTED** [review]
Using pixel values instead of dynamically changing variables can cause future problems.

**Comment by BaBrixx** [comment]
> Using pixel values instead of dynamically changing variables can cause future problems.

Can you elaborate on that @christian140398?

Pixels are used intentionally on the sidebar. Regarding the size of the top- and bottom nav bars, this is only a skeleton so the ones properly implementing them should change it.

In further reviews please mark the specific lines and write potential fixes.

**Review by atjn — CHANGES_REQUESTED** [review]
I agree with the general idea that using `em` instead of `px` will make our life easier in the future. But in certain cases, it is fine to use `px`, and one of them is the size of the sidepanels.

The way we are currently scaling the panels is the worst possible way to do it IMO. There are two things we need to optimize for:

### Initial size
This is the default size of the panels when the application is opened. If you define the size as a percentage of the screen, you will often find that the size is too small on small/lowres screens, and too large on large/highres screens. If you use pixels, the browser will change the relative size of the pixels to match the size that is right for the screen size and resolution.

### Zooming
This is the size of the panels after the browser has been zoomed. When zooming, a user generally does not want the size of the bars to change, but wants the content inside the panels to change size. If the size is percentage-based, it does not change when zoomed, but if it is pixel-based, it will change.

## Solutions

This is how the different solutions stack up:

### Current solution

We are currently using pixels, but are using percentage-based math to calculate the pixel size at startup.

Cons:
- Initial size
- Zooming
- (requires extra JavaScript to run when mounting)

### Solution using pixels

This is what we were doing before commit 768e686.

Pros:
- Initial size

Cons:
- Zooming

### Solution using percentages

In this case we only use `vw` as the unit, and will need to calculate the value manually with `cursorX / windowWidth` when the sidepanel is dragged.

Pros:
- Zooming

Cons:
- Initial size

**Comment by atjn** [comment]
Given that we are planning to make `ctrl`+`mousewheel` the way to zoom inside the model view, we could also implement our own zooming method inside the panels. That way, there is nowhere in the view where you can activate the default zoom system, and therefore the "solution using pixels" would have no cons.

**Review by atjn — APPROVED** [review]
ILVM

**Review by KamyaPA — APPROVED** [review]
The others says it looks good

**Pull request merged**